packages/workbox-window/src/Workbox.ts

```diff
@@ -104,7 +104,7 @@ class Workbox extends WorkboxEventTarget {
     }
 
     if (!immediate && document.readyState !== 'complete') {
-      await new Promise((res) => addEventListener('load', res));
+      await new Promise((res) => window.addEventListener('load', res));
```

Comment:
I hope it's not a problem here, but I know in the past using `window` has been an issue for people including this code in an SSR context. I don't think switching to `self` changes that though, so I'd rather wait and see if it's a problem before doing something like [I had to do here](https://github.com/w3c/IntersectionObserver/blob/d3eece49746574f43798d411d9fea2968251b075/polyfill/intersection-observer.js#L12-L15).

Reply:
This particular code is preceded by `document.readyState !== 'complete'` in the `if()` clause, which seems like it would already be problematic if we're talking about SSR.